microsoft/fhir-server
# Duplicated Search Parameters

## Description

This is a new PR for fixing Duplicated Search Parameters, before we had this old [PR](https://github.com/microsoft/fhir-server/pull/3334/files) which was reverted due not passing INT tests.

* [Conditional Create docs](https://www.hl7.org/fhir/http.html#ccreate)
* [Conditional Update docs](https://github.com/microsoft/fhir-server/pull/3334/files)
* [Update docs](https://github.com/microsoft/fhir-server/pull/3334/files)

The fix is pretty simple, but now I'm adding specific tests and also for each request I'm using a different [Code, Base], so if it fails again, we will know which is failing.

## Related issues
Addresses [issue #].

## Testing
Apart of testing, I used VS code rest to make sure no search parameters where created with the fix.

A simple way to test this is:

- Go to this branch `abigail/fix_duplicated_search_parameters`
- Go to `src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs` and delete lines 179, 214, and 232
- Run all tests added in ConditionalCreate, Conditional Update and Update.
- Make sure no duplicated search parameters are created. (For this I would prefer run a brand new server, and simple see that no new search parameters are created)

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- [ ] CI is green before merge [![Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_apis/build/status/CI%20Build%20%26%20Deploy?branchName=main)](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)
